lib/veri_report.ml

```diff
@@ -0,0 +1,56 @@
+open Core_kernel.Std
+open Bap.Std
+open Bap_traces.Std 
+
+module Map = String.Map
+
+type hist = (string * int) list [@@deriving bin_io, sexp]
+
+type succ = [ 
```

Comment:
Use `Result.t` type instead of success. First of all `Wrong` is hardly a success. Second, there is no need to reinvent the result type (and reinvent it wrongly). Third, you reinvention doesn't allow to discriminate between successful and unsuccessful results (according to the name they are all successes). 

And finally don't use unreadable and uninformative names like `Undsm`, especially if you're putting them in a public intreface.